src/main/java/org/jitsi/jicofo/FocusManager.java

```diff
@@ -686,11 +686,16 @@ private int getNonHealthCheckConferenceCount()
     /**
      * Sets the listener that will be notified about conference focus
      * allocation/disposal.
+     * @param className the listener classname.
      * @param l the listener instance to be registered.
      */
-    public void setFocusAllocationListener(FocusAllocationListener l)
+    public void setFocusAllocationListener(String className, FocusAllocationListener l)
```

Comment:
Is this className needed? Why don't you change the method to addFocusAllocationListener and drop the className param?

Reply:
instead of hashmap, you may use list. Every class can add itself or delete itself. like,

        chatRoom.addMemberPresenceListener(this); X  chatRoom.removeMemberPresenceListener(this);

Reply:
I replaced hashmap with list so everyone should be happy